pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/multithreading/xml/DoubleCheckedLocking.xml

```diff
@@ -128,12 +128,64 @@ public class Foo {
             synchronized (Foo.class) {
                 result = instance;
                 if (result == null) {
-                    result = instance = new Foo();
+                    instance = new Foo();
+                    result = instance;
```

Comment:
I'm unsure about this double assignment: The testcase only gave the expected result, because the rule was checking if exactly one assignment is present within the inner if block.

Should `result = instance = new Foo();` even have resulted in 0 expected problems, since `result` was not referencing the `static volatile instance` directly?

Any opinions?